File: 01_introduccion/ejemplos/calculo_maximo.py
def maximo(lista):
    max_valor = lista[0]
    for num in lista[1:]:
        if num > max_valor:
            max_valor = num
    return max_valor

# Uso de la función
numeros = [3, 5, 7, 2, 8, 1]
resultado = maximo(numeros)
print(f"El número máximo es: {resultado}")

File: 01_introduccion/ejemplos/calculo_maximo.hs
maximo :: [Int] -> Int
maximo [x] = x  -- Caso base: si solo hay un elemento, es el máximo
maximo (x:xs) = max x (maximo xs)  -- Comparamos la cabeza con el máximo de la cola

-- Uso de la función
main = do
    let numeros = [3, 5, 7, 2, 8, 1]
    print ("El número máximo es: " ++ show (maximo numeros))
